README


This archive contains the distribution HTML-FormFu,
version 2.01:

  HTML Form Creation, Rendering and Validation Framework

This software is copyright (c) 2013 by Carl Franks.

This is free software; you can redistribute it and/or modify it under
the same terms as the Perl 5 programming language system itself.


This README file was generated by Dist::Zilla::Plugin::Readme v5.015.



Changes
2.01 2014-05-05

    - Avoid uninitialized warning

2.00 2014-04-11

    - New layout() method for customizing output of fields.
      Any custom elements which override string() will likely need modified.

    - Deprecated: reverse_single() and reverse_multi() are deprecated,
      and warn when used.
      If the new layout() is used, and there is no simple way to replicate
      the behaviour, these methods will have no affect.

    - TT template files changed - update them if you use a local copy.
      Files updated: field
      New files: field_layout, field_layout_block,
        field_layout_checkboxgroup_field, field_layout_comment,
        field_layout_contentbutton_field, field_layout_errors,
        field_layout_field, field_layout_javascript, field_layout_label,
        field_layout_label_field, field_layout_label_text,
        field_layout_multi_field, field_layout_parser,
        field_layout_select_field, field_layout_textarea_field.
      Files deleted due to layout() changes: checkboxgroup_tag, 
        content_button, errors, input, input_tag, label, label_element,
        multi, select_tag, textarea_tag
      Unused files deleted: checkboxgroup.

    - New Filter::ForceListValue addresses rt bug #90813

    - render_label() and render_field() no longer require TT.
      Include render_label() in tests.

    - Fix typo in I18N::ja (Yusuke Watase).

    - Bundle our own (renamed) copy of MooseX::Attribute::Chained to avoid
      warnings under perl 5.19.x

1.00 2013-12-16

    - TT template files changed - update them if you use a local copy.
      Template file 'label_tag' renamed to 'label_element' - old file can
      be deleted. 'field' file changed. New 'errors' file.
    
    - TT no longer listed as a prerequisite. If you use the TT files,
      you must add 'Template' to your own app's prereqs.
    
    - Element::reCAPTCHA and Constraint::reCAPTCHA moved out to separate
      distribution.
    
    - HTML::FormFu::MultiForm moved out to separate distribution.
    
    - auto_container_class(), auto_label_class(), auto_comment_class(),
      auto_container_error_class(), auto_container_per_error_class(),
      auto_error_class() no longer have default values.
      See "RESTORING LEGACY HTML CLASSES" in HTML::FormFu docs to restore
      previous behaviour.
    
    - auto_label_class() no longer adds class to container.
      auto_label_class() now adds class to label tag.
      new auto_container_label_class() adds class to container.
      See "RESTORING LEGACY HTML CLASSES" in HTML::FormFu docs to restore
      previous behaviour.
    
    - auto_comment_class() no longer adds class to both container and comment.
      auto_comment_class() now only adds class to comment tag.
      new auto_container_comment_class() adds class to container.
      See "RESTORING LEGACY HTML CLASSES" in HTML::FormFu docs to restore
      previous behaviour.
    
    - Bug fix: param_value() form method now matches documented behaviour -
      returns undef when field has errors. (Reported by Hailin Hu).
    
    - New Element::Email and Element::URL HTML5 input fields.
    
    - Role::Element::Input has new datalist_options(), datalist_values(),
      datalist_id() and auto_datalist_id() methods to support HTML5 datalists.
      auto_datalist_id() is an inherited accessor which can be set on the 
      Form, MultiForm, or Block.
    
    - Form and Elements has new title() attribute short-cut.
    
    - Constraint::Regex has new anchored() accessor.
    
    - New Input attribute accessors: placeholder(), pattern(), autocomplete().
    
    - New Input boolean attribute accessors: autofocus(), multiple(), required().
    
    - New Field inherited accessors: auto_container_per_error_class(),
      auto_error_container_class(), auto_error_container_per_error_class(),
      error_tag(), error_container_tag
    
    - Constraints have new experimental method fetch_error_message().
    
    - All field elements have new method error_filename().
    
    - default_args() now supports 'Block', 'Field', 'Input' pseudo-elements,
      '|' alternatives, and '+' and '-' ancestor modifiers.
    
    - New Czech (cs) I18N translation by Jan Grmela.
    
    - mk_inherited_accessors() now also creates a *_no_inherit() method.
    
    - Experimental new roles() form method.
    
    - form methods start(), end() now respect render_method - no longer
      force use of tt templates.
    
    - Bug fix: del_attribute() on empty attribute no longer sets the attribute.
    
    - All attribute accessors generated with mk_attrs() now have *_loc variants.
    
    - form methods start(), end() now respect render_method - no longer
      force use of tt templates.
    
    - Tests now always require Test::Aggregate::Nested.
      Re-enable aggregate tests on Win32.
      Don't run all tests twice under both aggregate and t/ (doh!)

0.09010 2012-10-05

    - Internal changes - all Repeatable/nested_name munging is moved out of
      HTML::FormFu::Element::Repeatable into individual constraints

0.09009 2012-09-29
    - Make sure object can('checked') before calling checked() (colinnewell)
    - Updated Repeatable control to update id_field on DBIC::Unique if present
    - ComboBox new get_select_field_nested_name(), get_text_field_nested_name()
      accessors.
    - Fieldset new legend_attributes() method.
    - New form_error_message_class() method.
    - Constraint 'when' callback now receives $constraint as 2nd argument.

0.09007 2012-01-23
    - bump MooseX::Attribute::Chained version

0.09006 2012-01-23
    - fixed deprecation warnings of MX::Attribute::Chained (bricas)
    - Added placeholder attributes for types Text and Textarea with L10N support.
    - Added L10N support for 'prefix' attributes for types Date and DateTime.
    - Added 'attributes' support to types Date and DateTime.

0.09005 2011-09-06
    - bump version of prereq CGI to 3.37 to make all tests pass

0.09004 2011-08-26


Makefile.PL

# This file was automatically generated by Dist::Zilla::Plugin::MakeMaker v5.015.
use strict;
use warnings;

use 5.008001;

use ExtUtils::MakeMaker 6.30;

use File::ShareDir::Install;
install_share dist => "share";


my %WriteMakefileArgs = (
  "ABSTRACT" => "HTML Form Creation, Rendering and Validation Framework",
  "AUTHOR" => "Carl Franks <cpan\@fireartist.com>",
  "BUILD_REQUIRES" => {},
  "CONFIGURE_REQUIRES" => {
    "ExtUtils::MakeMaker" => "6.30",
    "File::ShareDir::Install" => "0.03"
  },
  "DISTNAME" => "HTML-FormFu",
  "EXE_FILES" => [
    "bin/html_formfu_deploy.pl",
    "bin/html_formfu_dumpconf.pl"
  ],
  "LICENSE" => "perl",
  "NAME" => "HTML::FormFu",
  "PREREQ_PM" => {
    "Carp" => 0,
    "Class::MOP::Method" => 0,
    "Clone" => "0.31",
    "Config::Any" => "0.18",
    "Cwd" => 0,
    "Data::Visitor" => "0.26",
    "Data::Visitor::Callback" => 0,
    "DateTime" => "0.54",
    "DateTime::Format::Builder" => "0.7901",
    "DateTime::Format::Natural" => 0,
    "DateTime::Format::Strptime" => "1.2000",
    "DateTime::Locale" => "0.45",
    "Email::Valid" => 0,
    "Encode" => 0,
    "Exporter" => "5.57",
    "Fatal" => 0,
    "File::Copy" => 0,
    "File::Find" => 0,
    "File::ShareDir" => 0,
    "File::Spec" => 0,
    "File::Temp" => 0,
    "HTML::Scrubber" => 0,
    "HTML::TokeParser::Simple" => "3.14",
    "HTTP::Headers" => "1.64",
    "Hash::Flatten" => 0,
    "IO::File" => 0,
    "List::MoreUtils" => 0,
    "List::Util" => 0,
    "Locale::Maketext" => 0,
    "Module::Pluggable" => 0,
    "Moose" => "1.00",
    "Moose::Role" => 0,
    "Moose::Util" => 0,
    "MooseX::Aliases" => 0,
    "Number::Format" => 0,
    "Path::Class::File" => 0,
    "Readonly" => 0,
    "Regexp::Common" => 0,
    "Scalar::Util" => 0,
    "Storable" => 0,
    "Task::Weaken" => 0,
    "YAML::XS" => "0.32"
  },
  "TEST_REQUIRES" => {
    "CGI" => "3.37",
    "POSIX" => 0,
    "Test::Aggregate::Nested" => "0.371",
    "Test::Exception" => 0,
    "Test::More" => "0.92"
  },
  "VERSION" => "2.01",
  "test" => {
    "TESTS" => "t/*.t"
  }
);


my %FallbackPrereqs = (
  "CGI" => "3.37",
  "Carp" => 0,
  "Class::MOP::Method" => 0,
  "Clone" => "0.31",
  "Config::Any" => "0.18",
  "Cwd" => 0,
  "Data::Visitor" => "0.26",
  "Data::Visitor::Callback" => 0,
  "DateTime" => "0.54",
  "DateTime::Format::Builder" => "0.7901",
  "DateTime::Format::Natural" => 0,
  "DateTime::Format::Strptime" => "1.2000",
  "DateTime::Locale" => "0.45",
  "Email::Valid" => 0,
  "Encode" => 0,
  "Exporter" => "5.57",
  "Fatal" => 0,
  "File::Copy" => 0,
  "File::Find" => 0,
  "File::ShareDir" => 0,
  "File::Spec" => 0,
  "File::Temp" => 0,
  "HTML::Scrubber" => 0,
  "HTML::TokeParser::Simple" => "3.14",
  "HTTP::Headers" => "1.64",
  "Hash::Flatten" => 0,
  "IO::File" => 0,
  "List::MoreUtils" => 0,
  "List::Util" => 0,
  "Locale::Maketext" => 0,
  "Module::Pluggable" => 0,
  "Moose" => "1.00",
  "Moose::Role" => 0,
  "Moose::Util" => 0,
  "MooseX::Aliases" => 0,
  "Number::Format" => 0,
  "POSIX" => 0,
  "Path::Class::File" => 0,
  "Readonly" => 0,
  "Regexp::Common" => 0,
  "Scalar::Util" => 0,
  "Storable" => 0,
  "Task::Weaken" => 0,
  "Test::Aggregate::Nested" => "0.371",
  "Test::Exception" => 0,
  "Test::More" => "0.92",
  "YAML::XS" => "0.32"
);


unless ( eval { ExtUtils::MakeMaker->VERSION(6.63_03) } ) {
  delete $WriteMakefileArgs{TEST_REQUIRES};
  delete $WriteMakefileArgs{BUILD_REQUIRES};
  $WriteMakefileArgs{PREREQ_PM} = \%FallbackPrereqs;
}

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

WriteMakefile(%WriteMakefileArgs);

{
package
MY;
use File::ShareDir::Install qw(postamble);
}


